test/unit/test-amp-story-player.js

```diff
@@ -370,5 +370,45 @@ describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
         'Story URL not found in the player: https://example.com/story6.html'
       );
     });
+
+    it('navigate to stories given a number', async () => {
+      const playerEl = win.document.createElement('amp-story-player');
+      appendStoriesToPlayer(playerEl, 5);
+
+      const player = new AmpStoryPlayer(win, playerEl);
+
+      await player.load();
+
+      const iframes = playerEl.shadowRoot.querySelectorAll('iframe');
+
+      player.go(0);
+      expect(iframes[0].getAttribute('i-amphtml-iframe-position')).to.eql('0');
```

Comment:
Did you try wrapping these inside a `win.requestAnimationFrame(() => {})` like the other tests to see if that helps?

https://github.com/ampproject/amphtml/blob/33e0c090956904a9adcf1d877b4c7727a6352543/test/unit/test-amp-story-player.js#L216-L220